test/parallel/test-tls-multi-key.js

```diff
@@ -29,15 +31,36 @@ var server = tls.createServer(options, function(conn) {
     ciphers: 'ECDHE-ECDSA-AES256-GCM-SHA384',
     rejectUnauthorized: false
   }, function() {
+    ecdsa.on('end', function() {
+      ecdsa.end();
+      end1 = true;
+      if (end2) {
+        server.close();
+      }
+    });
+
+    ecdsa.on('data', function(data) {
+      assert.equal(data, 'ok');
+    });
+
     ciphers.push(ecdsa.getCipher());
     var rsa = tls.connect(common.PORT, {
       ciphers: 'ECDHE-RSA-AES256-GCM-SHA384',
       rejectUnauthorized: false
     }, function() {
+      rsa.on('end', function() {
+        rsa.end();
+        end2 = true;
+        if (end1) {
+          server.close();
+        }
+      });
+
+      rsa.on('data', function(data) {
+        assert.equal(data, 'ok');
+      });
+
       ciphers.push(rsa.getCipher());
-      ecdsa.destroy();
```

Comment:
I think it should be safe to just replace `destroy` with `end`. Will the test pass with this?